### DISCUSSION TITLE
What tools are you using to develop this?

### DISCUSSION THREAD

**Discussion by steffenWi** [geefr/beatsaber-linux-goodies#76] [Q&A] [answered]
As a C# developer, Beat Saber player and Linux user, I was hoping I could help with this project. 

My main issue right now is that this doesn't seem to compile with Visual Studio Code which, as far as I know,  is the only C# IDE that currently runs on Linux and is available for free.

Your choice of UI framework hints at Jetbrain's Rider IDE, is that correct?

**Comment by Sebiann** [comment]
Im not one hundred percent sure, but doesn't Jetbrains IntelliJ IDEA have C# plugins?

**Reply by steffenWi** [reply]
It apparently doesn't. See [here](https://youtrack.jetbrains.com/issue/RIDER-2715) I'm a bit surprised by that actually. While I had forgotten IDEA existed, when you mentioned it I remembered it as a multi-language IDE. There is an extension for C# [here](https://plugins.jetbrains.com/plugin/7902-csharp) but that never went anywhere apparently and was published/last updated in 2015.

**Comment by geefr** [comment, marked as answer]
Woo! first proper question ;)

Don't have time for a full answer but here's what I'm using:

CI builds - That's github actions or circleci, dotnet container, just runs dotnet publish
On Linux - VSCode, dotnet sdk (5.0+), hand-editing the avalonia files
On Windows - VS 2019 with .net 5 SDK, avalonia designer extensions

Not using Jetbrains or anything of that ilk due to price/licencing model.
Generally if I'm doing UI design I'll do it on windows, as there's a visual designer extension for it - In vscode the best I could find was a preview pane for avalonia, but no widget toolbox (I'm a C++/Qt guy, don't know what all the widgets are called yet in C# world).

So what's your error? It definitely builds using the dotnet command line, nothing special needed (will pull avalonia from nuget, or whatever package manager dotnet uses)

**Reply by steffenWi** [reply]
Thanks for listing everything! Turns out, I hadn't properly  read the wiki for ArchLinux. To install the .NET Framework 5 SDK one needs to execute a script. The default dotnet-sdk package hasn't been updated yet.

The error I got said that the reference to .NETFramework 5 couldn't be found.

**Reply by geefr** [reply]
Ah Arch strikes again. As a lazy Linux user I recommend Mint/Ubuntu 20.04 - Can be fairly sure Microsoft will support these well, as they're a common choice in the professional world

**Reply by steffenWi** [reply]
Yes, I've noticed in the past that Arch may not be the best distribution to develop software on ;). Might set up a VM for coding, as Arch suits me perfectly for gaming.

**Reply by viasux** [reply]
Microsoft supports Ubuntu because they are run by a company and somewhat closed source (canonical's snap, for example) - which defeats the entire point of linux. The fact that ubuntu is company-run and their developers care more about money than quality is exactly why you should stay away.
